intermediate-goals.Rmd

```diff
@@ -0,0 +1,148 @@
+# Intermediate Outline {#intermediate-outline}
+
+This outline describes the questions that the intermediate course will answer.
+It assumes learners have completed the novice course, 
+which means they know:
+
+-   how to write code (in Python or R) that reads, analyses and plots a dataset,
+-   how to write and run unit tests,
+-   how to publish a static website using GitHub Pages,
+-   how to organize the files in a small project,
+-   the basics of licensing, and
+-   the basics of good behavior.
+
+## The Unix Shell
+
+Based on <http://swcarpentry.github.io/shell-novice/>
+and <https://merely-useful.github.io/still-magic/en/shell.html>
+
+-   How do I navigate my filesystem?
+    -   moving about (cd, ls, ., ..)
+    -   files and directories (ls, cp, mkdir, mv, rm)
+    -   paths (absolute, relative, special) 
+-   How do I examine and/or edit a text file?
+    -   difference between text and binary files
+    -   running a text editor
+    -   operating on text (cat, wc, head, tail, etc)
+    -   finding things (find, grep)
+-   How do I combine multiple commands?
+    -   pipes and filters
+    -   loops
+-   Variables
+    -   define a variable
+    -   use a variable
+    -   what's with all the quotes?
+-   Shell scripts
+    -   read/write/execute file permissions
+    -   `#!` lines
+    -   using command-line arguments in scripts
+-   Shortcuts
+    -   tab completion
+    -   previous commands (up arrow, control-r)
+    -   history
```

Comment:
Do you mean we should add the "finding things" content from http://swcarpentry.github.io/shell-novice/07-find/index.html

Reply:
I just meant mentioning piping history to grep since it is such a common pattern. But that could also be done when you introduce grep and find I think (line 27), I just want to make sure it is included since it is so helpful.